src/data/thermal-conductivity-constants.ts

```diff
@@ -2784,11 +3447,13 @@ export function calculateDIPPRThermalConductivity(
   let thermalConductivity = NaN;
   if (
     thermalConductivityProps !== undefined &&
-    temperature >= thermalConductivityProps.minimumTemperature &&
-    temperature <= thermalConductivityProps.maximumTemperature
+    temperature >= (thermalConductivityProps.minimumTemperature as number) &&
+    temperature <= (thermalConductivityProps.maximumTemperature as number)
```

Comment:
`as assertion` should not be used. Instead the function should be modified to handle `undefined` value.

Reply:
@maneetgoyal but sir , I had to make thermal conductivity at Tmax and Tmin optional . So I have to use number.isfinit() function with as assertion

Reply:
```js
export function calculateDIPPRThermalConductivity(
  compound: string,
  temperature: number
): number {
  const thermalConductivityProps: DIPPRThermalConductivityProps | undefined =
    DIPPRThermalConductivityDictionary[compound];
  let thermalConductivity = NaN;

  if (
    thermalConductivityProps !== undefined &&
    thermalConductivityProps.minimumTemperature !== undefined &&
    thermalConductivityProps.maximumTemperature !== undefined &&
    temperature >= thermalConductivityProps.minimumTemperature &&
    temperature <= thermalConductivityProps.maximumTemperature
  ) {
    thermalConductivity =
      (Number.isFinite(thermalConductivityProps.C1)
        ? (thermalConductivityProps.C1 as number)
        : 0) +
      (Number.isFinite(thermalConductivityProps.C2)
        ? (thermalConductivityProps.C2 as number) * temperature
        : 0) +
      (Number.isFinite(thermalConductivityProps.C3)
        ? Math.pow(temperature, 2) * (thermalConductivityProps.C3 as number)
        : 0) +
      (Number.isFinite(thermalConductivityProps.C4)
        ? Math.pow(temperature, 3) * (thermalConductivityProps.C4 as number)
        : 0) +
      (Number.isFinite(thermalConductivityProps.C5)
        ? Math.pow(temperature, 4) * (thermalConductivityProps.C5 as number)
        : 0);
  }

  return thermalConductivity;
}

```

This should work i think incase the `minimumTemperature ` or`maximumTemperature ` are undefined we return `NaN`.

Reply:
>does this suggestion seems okay to you?

Yeah it sounds good. The current implementation uses `typeof` check which is a bit more robust though.